accounts/abi/bind/topics.go

```diff
@@ -187,3 +187,52 @@ func parseTopics(out interface{}, fields abi.Arguments, topics []common.Hash) er
 	}
 	return nil
 }
+
+// parseTopics converts the indexed topic field-value pairs into map key-value pairs
+func parseTopicsIntoMap(out map[string]interface{}, fields abi.Arguments, topics []common.Hash) error {
+	// Sanity check that the fields and topics match up
+	if len(fields) != len(topics) {
+		return errors.New("topic/field count mismatch")
+	}
+	// Iterate over all the fields and reconstruct them from topics
+	for _, arg := range fields {
+		if !arg.Indexed {
+			return errors.New("non-indexed field in topic reconstruction")
+		}
+
+		switch arg.Type.T {
+		case abi.BoolTy:
+			if topics[0][common.HashLength-1] == 1 {
+				out[arg.Name] = true
+			} else {
+				out[arg.Name] = false
+			}
+		case abi.IntTy, abi.UintTy:
+			num := new(big.Int).SetBytes(topics[0][:])
+			out[arg.Name] = num
+		case abi.AddressTy:
+			var addr common.Address
+			copy(addr[:], topics[0][common.HashLength-common.AddressLength:])
+			out[arg.Name] = addr
+		case abi.HashTy:
+			out[arg.Name] = topics[0]
+		case abi.BytesTy, abi.FixedBytesTy:
+			out[arg.Name] = topics[0][:]
+		case abi.StringTy:
+			bytes := topics[0].Bytes()
+			var trimmedBytes []byte
+			for i, by := range bytes {
+				if by != 0 {
+					trimmedBytes = bytes[i:]
+					break
+				}
+			}
+			out[arg.Name] = string(trimmedBytes)
+		default:
```

Comment:
So after taking a closer look at the [docs](https://solidity.readthedocs.io/en/latest/contracts.html#events) I've realized that all array types (including `bytes` and `string`) do not have their byte-encoded values stored in the topic (even if the value fits in the 32 bytes) but rather have the keccak256 hash of their values stored in the topic. So in the case of these types the best we can do is retrieve said hash, which is what I've switched the implementation to here. This is still an improvement mostly because it means we can unpack event logs with these indexed types at all, whereas previously we would throw an error.